sdk/core/azure-core/tests/perf_tests/upload_binary.py

```diff
@@ -0,0 +1,80 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from time import time
+from wsgiref.handlers import format_date_time
+from devtools_testutils.perfstress_tests import RandomStream, AsyncRandomStream
+
+from azure.core.rest import HttpRequest
+from azure.core.exceptions import (
+    HttpResponseError,
+    map_error,
+)
+from ._test_base import _BlobTest
+
+
+class UploadBinaryDataTest(_BlobTest):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        blob_name = "uploadtest"
+        self.blob_endpoint = f"{self.account_endpoint}{self.container_name}/{blob_name}"
+        self.upload_stream = RandomStream(self.args.size)
+        self.upload_stream_async = AsyncRandomStream(self.args.size)
+
+    def run_sync(self):
+        self.upload_stream.reset()
+        data = self.upload_stream.read(self.args.size)
+        current_time = format_date_time(time())
+        request = HttpRequest(
+            method="PUT",
+            url=self.blob_endpoint,
+            params={},
+            headers={
+                'x-ms-date': current_time,
+                'x-ms-blob-type': 'BlockBlob',
+                'Content-Length': str(self.args.size),
+                'x-ms-version': self.api_version,
+                'Content-Type': 'application/octet-stream',
+                'Accept': 'application/octet-stream',
+            },
+            content=data
+        )
+        response = self.pipeline_client._pipeline.run(
+            request,
+            stream=True
```

Comment:
I _think_ `stream=True` is for streaming the response, not the request - but I could be wrong...

Reply:
ah yep you are right, just checked [docs](https://github.com/Azure/azure-sdk-for-python/blob/370ecaa81d4cc4d0ec33430855cc48c8394073e5/sdk/core/azure-core/azure/core/_pipeline_client.py#L193)! removing